### PR TITLE
Add a `wrapJSExceptions()` function in a new JS-specific library

### PR DESCRIPTION
**Pull request by nex3**
This encapsulates a workaround for dart-lang/sdk#53105.

### CODE REVIEW

File: lib/js.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+/// JS context should be wrapped in this function.
+///
+/// [a bug]: https://github.com/dart-lang/sdk/issues/53105
+/// [strict mode]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode
+T wrapJSExceptions<T>(T Function() callback) {
+  if (!_isStrictMode) return callback();
+
+  try {
+    return callback();
+  } on String catch (error) {
```
Comment (Goodwine) [inline lib/js.dart:45]: `null` and `undefined` are also primitives and should be handled as well, or at least they should have tests to verify they don't need the handling

File: test/npm_test.dart [hunk 2]
```diff
@@ -1022,6 +1070,66 @@ void main() {
       await process.shouldExit(1);
     });
   });
+
+  group("package:cli_pkg/js.dart", () {
+    group("wrapJSException in strict mode", () {
+      // Asserts that the JS [expression] doesn't crash when caught by Dart code
+      // after going through `wrapJSExceptions()`.
```
Comment (Goodwine) [inline test/npm_test.dart:1077]: Should this use `///`?

**Review by Goodwine — CHANGES_REQUESTED** [review]

File: test/npm_test.dart (hunk 2 above)
Comment (nex3) [reply test/npm_test.dart:1077]: No, this is a local variable so DartDoc will ignore it regardless.

Comment (Goodwine) [reply test/npm_test.dart:1077]: TIL nested functions don't get dartdoc even for IDEs
https://github.com/dart-lang/sdk/issues/39746

**Review by Goodwine — APPROVED** [review]

**Pull request merged**